### PR TITLE
updated hue and added config for hue and spark

### PR DESCRIPTION
**Pull request by antonkulaga**
I updated hue to latest 4.0 versions and added volumes to configure spark and hue.

### CODE REVIEW

File: conf/spark/master.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# increased timeouts as defaults value is too low for many applications and lead to crashes
+spark.network.timeout             1024s
```
Comment (earthquakesan) [inline conf/spark/master.conf]: Hi @antonkulaga!

In general, it is better to make components configurable via environmental vars. Configuration through mounting the files like this tends to be cumbersome and it is also hard to follow. In case of env vars, all the config is kept in docker-compose or separate env file.

File: docker-compose-hive.yml [hunk 2]
```diff
@@ -64,8 +66,11 @@ services:
     ports:
       - 9001:9001
   hue:
-    image: bde2020/hdfs-filebrowser:3.11
+    image: gethue/hue:latest
```
Comment (earthquakesan) [inline docker-compose-hive.yml]: I remember now the reasons for not using official hue docker image:
1. There is no Dockerfile provided, therefore I can not see how secure it is or what is installed inside.
2. I can not extend it easily as there is only 1 "latest" tag. Any extension will mean that it will break in the future.
3. No configuration via env variables.

I will update the hdfs-filebrowser in the next days and provide more information on how to use it (in case you need all the applications and not only filebrowser).

Comment (antonkulaga) [inline docker-compose-hive.yml]: >There is no Dockerfile provided

Do not they use docker file in hue github repo?

>I can not extend it easily as there is only 1 "latest" tag

Agree on that

File: conf/spark/master.conf (hunk 1 above)
Comment (antonkulaga) [inline conf/spark/master.conf]: @earthquakesan can you put a code that generates environment variable for all configuration properties? For instance it is not clear for me how I can switch off and on hue apps with enviroment variables with your hue docker. I also have to tweak other settings in hue.ini file depending on deployment (like reverse proxy, differen timeouts and so on), I have no clue how I can do it with your image and enviroment variables

**Comment by earthquakesan** [comment]
I close this pull request due to long period of inactivity.
